src/proxy/buffer.rs

```diff
@@ -236,11 +227,10 @@ where
     D: Deadline<Req>,
     Req: Send + 'static,
 {
-    pub fn new<E>(svc: S, deadline: D, capacity: usize, exec: &mut E) -> Self
-    where
-        E: buffer::WorkerExecutor<Dequeue<S>, Stealer<Req>>,
-    {
-        let inner = buffer::Buffer::with_executor(Dequeue(svc), capacity, exec);
+    pub fn new(svc: S, deadline: D, capacity: usize) -> Self {
+        let mut exec =
+            tokio::executor::DefaultExecutor::current().instrument(tracing::Span::current());
```

Comment:
Nit: if we update to `tracing-futures` 0.1, this can be:
```suggestion
            tokio::executor::DefaultExecutor::current().in_current_span();
```

Reply:
Also, I'm not sure if this is necessary — I think `tower-buffer` already propagates spans: https://github.com/tower-rs/tower/blob/e414b2b7d3ce8220aaf321d42af70cf37ec127d0/tower-buffer/src/service.rs#L96-L99